### PR TITLE
using find_package(Threads) for setting the compilation flags

### PR DESCRIPTION
**Pull request by adam852**
this is the solution for an issue(#360): "error: argument unused [...] '-pthread' "

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-PROJECT(binaryen CXX)
+PROJECT(binaryen C CXX)
```
Comment (jfbastien) [inline CMakeLists.txt]: Is C needed?

**Comment by jfbastien** [comment]
Bots fail with:
```
clang: error: -lpthread: 'linker' input unused
: error: -lpthread: 'linker' input unused
```

File: CMakeLists.txt (hunk 1 above)
Comment (adam852) [inline CMakeLists.txt]: Unfortunately C is needed for support of old version of cmake module FindThreads. Now cmake miniumum version is 2.8.7, from 3.4.0 it does'nt need C.

**Comment by kripken** [comment]
Looks like this passes on the bots now, is it ready?

**Comment by adam852** [comment]
Yes, it is.

**Comment by kripken** [comment]
Ok great, looks good to me to land. Have you already joined the [w3c wasm group](https://www.w3.org/community/webassembly/participants)? (see [here](https://github.com/WebAssembly/design/blob/master/Contributing.md))

**Comment by adam852** [comment]
I've just joined the w3c wasm group.

**Comment by kripken** [comment]
Excellent, thanks!

**Pull request merged**